src/gwt/src/org/rstudio/core/client/dom/DomUtils.java

```diff
@@ -895,4 +895,16 @@ public static final native void setSelectionRange(Element el, int start, int end
       if (el.setSelectionRange)
          el.setSelectionRange(start, end);
    }-*/;
+   
+   public static final native void copyCodeToClipboard(String text) /*-{
```

Comment:
Validated this in Safari + native OS X and IE 10+, looking good. Here is the full matrix: https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand - Therefore, we should be good, but I'll double check this and other features in all our platforms once this dialog is complete.